File: totosuki/ABC/B/293.py
N = int(input())
A = list(map(int, input().split()))
l = [False for _ in range(N)]
rslt = []
for i in range(N):
  if l[i] == False:
    l[A[i]-1] = True
for i in range(N):
  if l[i] == False:
    rslt.append(i+1)
print(len(rslt))
print(*rslt)

File: OCH_Tao/ABC/B/293.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const N = parseInt(input[0], 10);
const A = input[1].split(" ").map(x => parseInt(x, 10));
let S = new Set([...Array(N).keys()].map(x => x + 1));
for (let i = 0; i < N; i++) {
  if (S.has(i + 1)) {
    S.delete(A[i]);
  }
}
console.log(S.size);
console.log([...S].sort((a, b) => a - b).join(" "));